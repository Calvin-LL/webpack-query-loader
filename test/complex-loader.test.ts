import webpack from "webpack";

import compile from "./helpers/compile";
import execute from "./helpers/execute";
import getCompiler from "./helpers/getCompiler";
import readAsset from "./helpers/readAsset";

describe("complex loaders", () => {
  it("should work when multiple loaders with pitch stage is present", async () => {
    const compiler = getCompiler({}, false, "complex.js");
    const stats = await compile(compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
    expect((stats as webpack.Stats).compilation.warnings).toMatchSnapshot(
      "warnings"
    );
  });
});
