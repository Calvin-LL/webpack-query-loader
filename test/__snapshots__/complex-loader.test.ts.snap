// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`complex loaders should work when multiple loaders with pitch stage is present: result 1`] = `
Object {
  "data": Array [
    116,
    101,
    115,
    116,
  ],
  "type": "Buffer",
}
`;

exports[`complex loaders should work when multiple loaders with pitch stage is present: warnings 1`] = `
Array [
  [ModuleWarning: Module Warning (from <rootDir>/dist/cjs.js):
loader1-main test],
  [ModuleWarning: Module Warning (from <rootDir>/dist/cjs.js):
loader1-pitch <rootDir>/dist/cjs.js??ref--5-1!<rootDir>/test/fixtures/test.txt?test  {}],
  [ModuleWarning: Module Warning (from <rootDir>/dist/cjs.js):
loader2-main test],
  [ModuleWarning: Module Warning (from <rootDir>/dist/cjs.js):
loader2-pitch <rootDir>/test/fixtures/test.txt?test <rootDir>/dist/cjs.js??ref--5-0 {}],
]
`;
